src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
 'use strict';
+const wall = document.querySelector('.wall');
 
-document.addEventListener('click', (e) => {
-  // write code here
+wall.addEventListener('click', (e) => {
+  const spider = document.querySelector('.spider');
+  const spiderRect = spider.getBoundingClientRect();
+  const wallRect = wall.getBoundingClientRect();
+  const clickX = e.clientX - wallRect.left;
+  const clickY = e.clientY - wallRect.top;
+ 
+  let topPosition = clickY - (spiderRect.height / 2);
+  let leftPosition = clickX - (spiderRect.width / 2);
+
+  if (topPosition < 0) {
+    topPosition = 0;
+  } else if (topPosition + spiderRect.height > wallRect.height) {
+    topPosition = wallRect.height - spiderRect.height;
+  }
```

Comment:
This block of code is checking if the spider is within the wall boundaries. It would be more readable if you extracted this logic into a separate function, like `keepSpiderInBounds()`. This function could take the calculated position and the wall and spider dimensions as parameters and return the corrected position.